lib/nvme/nvme_internal.h

```diff
@@ -69,6 +69,12 @@
  */
 #define NVME_INTEL_QUIRK_WRITE_LATENCY 0x2
 
+/*
+ * The controller needs a delay before starts checking the device
+ * readiness, which is done by reading the NVME_CSTS_RDY bit.
```

Comment:
Can you describe in more detail what actually goes wrong?

Is it sufficient to just add a longer timeout to the WAIT_FOR_READY states, or does the actual reading of the CSTS register cause the problem?